buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/Keyword.java

```diff
@@ -24,34 +24,50 @@
  */
 public class Keyword {
   public final String keyword;
+  public final String value;
   public final String filePath;
 
   public Keyword(String keyword) {
-    this(keyword, /*filePath=*/ null);
+    this(keyword, keyword, /*filePath=*/ null);
+  }
+
+  public Keyword(String keyword, String value) {
+    this(keyword, value, /*filePath=*/ null);
   }
 
   /**
    * Creates a {@code Keyword}.
    *
-   * @param keyword The keyword string
+   * @param keyword The name of the keyword
+   * @param value The value of the keyword
    * @param filePath The file where this keyword was taken from
    */
-  public Keyword(String keyword, String filePath) {
+  public Keyword(String keyword, String value, String filePath) {
     this.keyword = Objects.requireNonNull(keyword.toUpperCase());
+    this.value = Objects.requireNonNull(value);
     this.filePath = filePath;
   }
 
   @Override public int hashCode() {
-    // The filePath should not be considered when calculating hashCode.
```

Comment:
Out of interest, what is the benefit of overriding the default hashCode function?

Reply:
I believe that the benefit is that Keyword objects will now be more efficiently used within a hashing datastructure (such as HashMap).

Reply:
More efficiently than the default hashcode?

Reply:
Oh, nvm, I think I understand. You are overriding the equals method as well, so I guess the default hashcode wouldn't work for this.